### PR TITLE
Add support for bare multisig

### PR DESCRIPTION
**Pull request by rubensayshi**
code is more or less copy paste of P2SH multisig code with some changes.

also added `normalizeSignatures` to take existing signatures (as buffer, from a partially signed TX for example) and turn them into `TransactionSignature` instances, not sure what a good name would be, don't think `normalizeSignatures` is...

### CODE REVIEW

File: lib/transaction/input/multisig.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+};
+
+MultiSigInput.OPCODES_SIZE = 7; // serialized size (<=3) + 0 .. N .. M OP_CHECKMULTISIG
+MultiSigInput.SIGNATURE_SIZE = 74; // size (1) + DER (<=72) + sighash (1)
+MultiSigInput.PUBKEY_SIZE = 34; // size (1) + DER (<=33)
+
+MultiSigInput.prototype._estimateSize = function() {
+  return MultiSigInput.OPCODES_SIZE +
+    this.threshold * MultiSigInput.SIGNATURE_SIZE +
+    this.publicKeys.length * MultiSigInput.PUBKEY_SIZE;
```
Comment (braydonf) [inline lib/transaction/input/multisig.js]: Would the estimated size include the public keys? Here is an example created with this code: https://test-insight.bitpay.com/tx/dd387bbdeaef66f25b1f1c876df98b1fc6fba62226c01af377800df820bfd338:

```
<Script: OP_0 72 0x3045022100e633c8ccafde4e3031ea5ced9666336b289f8cf64b6045a879afbd994b824fb502203d16d700e3efd0d957d17941c7b06e7050a9a0132b533742d4e016780d5f1d7001 72 0x30450221008f5dc4a27804b98f00cac764c05b026e49998e7ad469061fd67dc58ecb353acb022012f4b9be8d2a8bf03eed1d01a50c1891d6e691c4f19e9bca7227d8515a4db90901>
```
And it looks like there are only signatures.

**Comment by braydonf** [comment]
Went through and tested the full process from sending to a multisig output and then signing it, and looks to be working properly.
- out - https://test-insight.bitpay.com/tx/ca0e2b0c285d9db8cb1e9fe9cfe25d534914b4cbb5b8bacd1a45778616bd8b81
- signed - https://test-insight.bitpay.com/tx/dd387bbdeaef66f25b1f1c876df98b1fc6fba62226c01af377800df820bfd338

Looks like there may be some issues regarding the insight-ui display of these inputs/outputs, however unrelated to this PR.

**Comment by rubensayshi** [comment]
ugh somehow missed your comments ...

File: lib/transaction/input/multisig.js (hunk 1 above)
Comment (rubensayshi) [inline lib/transaction/input/multisig.js]: I'll take a look at it and add a test for it and make sure it works properly

**Comment by rubensayshi** [comment]
fixed `_estimateSize()` and added test

**Comment by braydonf** [comment]
LGTM

**Comment by matiu** [comment]
LGTM

**Comment by robby-dermody** [comment]
+1 - counterparty (counterwallet) will make good use of this feature. Looking forward to having it merged.

**Comment by rubensayshi** [comment]
awesome!

@braydonf will there be a tagged release soon with this?

**Comment by braydonf** [comment]
Landed in 0.13.10

**Pull request merged**